app/src/components/validators.js

```diff
@@ -162,7 +163,7 @@ const models = {
 
     /** @function from */
     from: value => {
-      return validatorUtils.isEmail(value);
+      return validatorUtils.isEmail(value) && !(config.get('server.blockDoNotReplySender') && (value.match(/^no[-]?reply/gi) || value.match(/^donotreply/gi)));
```

Comment:
Extending from the above comment, if you were testing for the existence of the value instead, you can instead do the following:
```js
if (config.has('server.blockDoNotReplySender')) { // do stuff }
```
I believe the regexes can be condensed together better... try the following regex:
```js
value.match(/^(do)?no(t|-)?reply/gi)
```
Lastly, it may not be a bad idea to functionalize the config check and regex match into its own `validatorUtils.blockDoNotReplySender` function instead as that'll keep it better scoped.